doc/examples/example_asymm_crypto.md

```diff
@@ -476,3 +476,101 @@ mococrw::EdDSASignaturePrivateKeyCtx ctx(key);
 // Hash of message is calculated automatically
 signature = ctx.signMessage(message);
 ```
+
+# Dilithium
+
+## Background
+Quantum computers can break RSA and elliptic curve based encryption using Shor's algorithm. However, nowadays, the number of available qubits in modern quantum computers does not suffice to successfully break RSA and ECC keys. But it is just a matter of time until enough qubits are available.
+
+CRYSTALS-Dilithium is a lattice-based signature scheme. There are no known quantum computer algorithms that can break dilithium.
+
+For further information regarding dilithium please refer to https://pq-crystals.org/dilithium/index.shtml .
+
+## Note on hashing algorithms
+Dilithium differs compared to the other algorithms. The other signing-/verification-algorithms described here expect a hashing algorithm when verifyMessage(...) or signMessage(...) is used. Dilithium does not.
+
+This behavior is because the dilithium signature algorithm can take arbitrary length inputs and internally digests them to obtain a fixed-length signature. Consequently, messages may be directly passed into it, or a hash algorithm of choice (e.g., SHA3-512) can be used to digest the message and then pass it into the verifyMessage interface here. This can be an implementation benefit as the dilithium interface doesn't support streaming for large amounts of data but only to pass all data in one go.
```

Comment:
Nit: Consider sentence restructure. "is because" does not really flow.

Maybe: "This behaviour arises as the dilithium...."

Reply:
Done